### ISSUE TITLE
Kodi crashes watching Sat-TV

### ISSUE THREAD

**Issue by ghost** [xbmc/xbmc#17531]
### Describe the bug
Here is a clear and concise description of what the problem is:

I just  watched Sat-TV. 
During the playback I noticed recurrently (every ~30-60 s) artefacts mostly on the moving parts of the pictures. After ~ 30 minutes the box crashed.
The CPU was on 6 up to  30 %.
 
maybe relevant:
if I run tail -f .kodi/temp/kodi.log with Debug on I see this:
...
2020-03-21 14:36:43.543 T:1632 WARNING: VAAPI::FFGetBuffer - no surface available - dec: 1, render: 2
2020-03-21 14:36:43.543 T:1632   ERROR: ffmpeg[0x4987080X]: [h264] get_buffer() failed
2020-03-21 14:36:43.543 T:1632   ERROR: ffmpeg[0x4987080X]: [h264] thread_get_buffer() failed
2020-03-21 14:36:43.543 T:1632   ERROR: ffmpeg[0x4987080X]: [h264] decode_slice_header error
2020-03-21 14:36:43.543 T:1632   ERROR: ffmpeg[0x4987080X]: [h264] no frame!
2020-03-21 14:36:43.593 T:1632   ERROR: ffmpeg[0x4987080X]: [h264] mmco: unref short failure
2020-03-21 14:36:43.594 T:1632   ERROR: ffmpeg[0x4987080X]: [h264] number of reference frames (0+5) exceeds max (4; probably corrupt input), discarding one
2020-03-21 14:36:43.641 T:1632   DEBUG: CPtsTracker: pattern lost on diff 40000.000000, number of losses 6
2020-03-21 14:36:46.093 T:1632   DEBUG: CPtsTracker: detected pattern of length 1: 20000.00, frameduration: 20000.000000
2020-03-21 14:36:59.060 T:1632   DEBUG: CPtsTracker: pattern lost on diff 40000.000000, number of losses 7
2020-03-21 14:36:59.092 T:1632   ERROR: ffmpeg[0x4987080X]: [h264] mmco: unref short failure
2020-03-21 14:36:59.092 T:1632   ERROR: ffmpeg[0x4987080X]: [h264] number of reference frames (0+5) exceeds max (4; probably corrupt input), discarding one
2020-03-21 14:37:01.593 T:1632   DEBUG: CPtsTracker: detected pattern of length 1: 20000.00, frameduration: 20000.000000
...

### To Reproduce

1. watch TV 
2. wait ~ 30 minutes

### Debuglog

The debuglog can be found here:
http://ix.io/2eQl

### Your Environment
Used Operating system:

 - [x] Linux

the box is an Intel NUC 8i3BEH (~3 month old) and can do sse4.1 in opposite what's in the crash log

 - Operating system version/name:
 - Kodi version: 19.0-ALPHA1 Git:4226dc1 (Milhouse Nightly #0320)
I also see/saw artefacts and crashes in LE 9.2 and an clean installed 9.2.1

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by DaVukovic** [comment]
@fritsch @MilhouseVH any idea? It seems to crash in libva.

**Comment by MilhouseVH** [comment]
Not really. I can upload a debug-enabled build later, if that's of any help to anyone (ie. reproduce the crash, this time with more details stack trace).

**Comment by MilhouseVH** [comment]
I've uploaded a debug-enabled build, [`#0321d`](http://milhouse.libreelec.tv/builds/debug/Generic/LibreELEC-Generic.x86_64-9.80-Milhouse-20200322023951-%230321d-gae22e11.tar)

@sixpack58: It will require a 1GB /flash partition - if your system has a 512MB /flash partition you will need to increase it by resizing the partition.

assuming you have a 1GB /flash partition you should be able to upgrade to this build from your current `#0320`. Reproduce the crash and upload the crash log which should now be much more detailed. You can then return to using your normal (non-debug) test build.

**Comment by sixpack58** [comment]
thanks for support and looking into this bug !!!
#0321d is running.
now, trying to get some usefull traces....

P.S.
before installing #0321d I was watching TV for 2-3 h with "artefacts" (?), but no crashes.

**Comment by sixpack58** [comment]
I would like to provide another log file. it's clearly not an crash log, but I guess it's anyhow from interesting, cause during the first 10 minutes watching TV I got some black screens the way that over ~ 5 minutes the TV picture came and go and changed into an black screen.
starting at ~ "2020-03-22 03:40:44.494" 
- but I'm not an experienced log file reader -
investigating further and waiting for an crash ...

the **link to the log**:
https://paste.ubuntu.com/p/BM9XS4bnjV/

**Comment by sixpack58** [comment]
I mentioned it today: 
why is this labeled with "v19 matrix" only ?
the crashes happen in "v18 Leia" (exact18.6) too !

**Comment by sixpack58** [comment]
it seems I found the culprit.

running TvHeadEnd from here:
https://forum.libreelec.tv/thread/3416-tvheadend-nightly-builds-for-le/?pageNo=25#post141374
or
http://build.mycvh.de/Tvh-LE9.2/

on LE 9.2.3 fixes the bug.
no crashes and very few artefacts (~1 in 60 minutes on 1 or 2 TV-channels)

I still need to to test on LE-nightly.

**Comment by sixpack58** [comment]
a new crash/debug log with Milhouse nightly #0626 watching Sat-TV
- _moons ago I read something regarding  "expressiver" debug logs. Maybe this is the one to nail this bug down (?!)_ -

https://paste.ubuntu.com/p/DVynjsWxkx/

P.S.
currently I have no TvHeadend 4.3 for LE-nightly to test with.
- _the above TvH 4.3 refused to install under nightly and it seems (to me) it's not easy to (temp.) circumvent it by just copy over the addon.xm_l  -
